src/v2/Apps/Collect/Routes/Collect/index.tsx

```diff
@@ -57,90 +54,102 @@ export const CollectApp = track({
-        <Title>{title}</Title>
-        <Meta property="og:url" content={`${sd.APP_URL}/collect`} />
-        <Meta
-          property="og:image"
-          content={`${sd.APP_URL}/images/og_image.jpg`}
-        />
-        <Meta name="description" content={description} />
-        <Meta property="og:description" content={description} />
-        <Meta property="twitter:description" content={description} />
-        <Link rel="canonical" href={canonicalHref} />
-
-        <BreadCrumbList items={items} />
-
-        {filterArtworks && <SeoProductsForArtworks artworks={filterArtworks} />}
-
-        <Box mt={3}>
-          <Serif size="8">
-            <h1>Collect art and design online</h1>
-          </Serif>
-          <Separator mt={2} mb={[2, 2, 2, 4]} />
-
-          <CollectionsHubsNav
-            marketingHubCollections={props.marketingHubCollections}
-          />
-
-          <Spacer mb={2} mt={[2, 2, 2, 4]} />
-        </Box>
-
-        <Box>
-          <ArtworkFilter
-            viewer={viewer}
-            filters={location.query as any}
-            sortOptions={[
-              { value: "-decayed_merch", text: "Default" },
-              { value: "-partner_updated_at", text: "Recently updated" },
-              { value: "-published_at", text: "Recently added" },
-              { value: "-year", text: "Artwork year (desc.)" },
-              { value: "year", text: "Artwork year (asc.)" },
-            ]}
-            onArtworkBrickClick={artwork => {
-              tracking.trackEvent(
-                clickedMainArtworkGrid({
-                  contextPageOwnerType: OwnerType.collect,
-                  destinationPageOwnerId: artwork.internalID,
-                  destinationPageOwnerSlug: artwork.slug,
-                }) as any
-              )
-            }}
-            onChange={filters => {
-              const url = buildUrlForCollectApp(filters)
-
-              if (typeof window !== "undefined") {
-                // FIXME: Is this the best way to guard against history updates
-                // in Storybooks?
-                if (!process.env.IS_STORYBOOK) {
-                  window.history.replaceState({}, "", url)
-                }
-              }
-
-              /**
-               * FIXME: Ideally we route using our router, but are running into
-               * synchronization issues between router state and URL bar state.
-               *
-               * See below example as an illustration:
-               *
-                const newLocation = router.createLocation(url)
-
-                router.replace({
-                  ...newLocation,
-                  state: {
-                    scrollTo: "#jump--artworkFilter"
-                  },
-                })
-               *
-               */
-            }}
-          />
-        </Box>
-      </FrameWithRecentlyViewed>
-    </AppContainer>
+    <AnalyticsContext.Provider
+      value={{
+        contextPageOwnerType: OwnerType.collect,
+      }}
```

Comment:
One thought here is that rather than wrap each individual app that needs analytics context values with a provider / consumer, we do it once in `Boot` and then _update_ the values via a hook that calls out the context.

Meaning:

```jsx
const MyApp = props => {
  const {contextPageOwnerType} = useAnalyticsContext({
    contextPageOwnerType: "foo",
    ...
  })

  return ...
}
```
Invoking the `useAnalyticsContext` hook with an argument updates the context with new values. Calling it bare returns the values that are there.

Then, for additional apps / new code, we only have to worry about the hook and avoid all the setup paperwork.

Reply:
Yes I definitely think the provider should be set higher in the chain, this was why I asked about possibility of linking it to the router. I'm hoping to do a follow up PR that will add context to all apps, want to pair on adding this later?

Reply:
Yup down to pair on this later (after week vacation starting tomorrow) 👍 

In that case everything in this PR looks great, all the building blocks are in place.